components/calculator.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Buttons from './buttons';
+import Display from './display';
+import calculate from '../logic/calculate';
+
+// eslint-disable-next-line react/prefer-stateless-function
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    const {
+      previous,
+      operand,
+      current,
+    } = props;
+    this.state = {
+      total: previous,
+      operation: operand,
+      next: current,
+    };
+    this.update = this.update.bind(this);
+  }
+
+  update(btnClick) {
+    const obj = calculate(this.state, btnClick);
+    this.setState(() => ({
+      total: obj.total,
+      operation: obj.operation,
+      next: obj.next,
+    }));
+  }
+
+  render() {
```

Comment:
Great job @amklass11 🤝

- [ ] I seem to get this error when I try to run the application. I couldn't figure out why. You may kindly fix this error and resubmit your work for review.
![error](https://user-images.githubusercontent.com/63917725/219576159-c8c5f850-7695-4dcf-8d4e-d8961d96e01a.PNG)